src/plugins/intel_cpu/src/node.h

```diff
@@ -633,6 +639,7 @@ class Node {
     std::vector<MemoryPtr> internalBlobMemory;
     std::vector<NodeDesc> supportedPrimitiveDescriptors;
     std::unordered_map<int, dnnl::memory> primArgs;
+    MemoryPtr scratchpadMem;
```

Comment:
It does not make sense to store the memory object inside the node object, since the memory inside scratch pad is constantly changing. So it may lead to misusing. As far as I understand, the only place where we really need to store the memory pointer, is RNN node, this in that case the args map is filled every exec call. I suggest storing dnnl::memory object inside RNN node object to reuse it in exec call. In all the other cases, accessing scrutchPad is just enough.

Reply:
each `scratchpadMem` object here will be registered into `DnnlMemoryMngr` inside scratchPad object so that any future scratch pad resize can trigger `DnnlMemoryMngr::notifyUpdate` which update the registered memory through `Memory::update` methods, so this object has to be alive even after it has produced the memory primitive in `primArgs`, we tried make it a temporary, but segmenfault happens in testing of real models.

Reply:
Hi, @maxnick : just as Tingqian mentioned, if we use the independent temporary variable way of 'MemoryPtr scratchpadMem' for each nodes. segment fault can be reproduced by 'data/cv_bench_cache/sk_13sept_75models_22.2_int8/resnet-50/' model.

Reply:
Given the explanations and the fact that it is better to couple scratch pad initialization steps in one place, what about the follows:
0. Move scratchpadMem to the `private` section of the base class.
1. Introduce a protected method:
```cpp
MemoryPtr Node::getScratchPadMem(const const_dnnl_primitive_desc_t& pd) {
        auto scratchpadMemoryDesc = DnnlExtensionUtils::query_md(pd, dnnl::query::scratchpad_md);
        scratchpadMem = getRuntimeScratchPad()->getScratchPadMem(scratchpadMemoryDesc);
        return scratchpadMem;
}
```
In such design we couple scratchpadMem usage with its initialization that would prevent usage of uninitialized or wrongly initialized scratchpad memory object.
3. RNN is special case here, since its ArgsMap is filled every exec call, in that case, I think it does make sense to store one more pointer to the scratchpadMem inside the RNN node object, to avoid extra reinitialization at the execute stage.
What do you think?

Reply:
looks great, have changed to this way.

Reply:
Thanks.